symbology/custom-dictionary-style/src/main/java/com/esri/samples/custom_dictionary_style/CustomDictionaryStyleSample.java

```diff
@@ -67,31 +91,124 @@ public void start(Stage stage) {
-      DictionarySymbolStyle restaurantStyle = DictionarySymbolStyle.createFromFile(styleFilePath);
-
-      // create a dictionary renderer
-      DictionaryRenderer dictionaryRenderer = new DictionaryRenderer(restaurantStyle);
-
-      // set the dictionary renderer to the restaurant layer
-      restaurantLayer.setRenderer(dictionaryRenderer);
-
-      // add the map view and control panel to stack pane
-      stackPane.getChildren().add(mapView);
+      // set up the UI
+      setupUI();
+
+      // enable UI interactions once the feature layer has loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          controlsVBox.setDisable(false);
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // create a portal item using the portal and the item id of the dictionary web style
+      Portal portal = new Portal("https://arcgisruntime.maps.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "adee951477014ec68d7cf0ea0579c800");
+
+      // create a stylex file from a local location
+      File stylxFile = new File(System.getProperty("data.dir"), "./samples-data/stylx/Restaurant.stylx");
+
+      // create a service feature table
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/Redlands_Restaurants/FeatureServer/0");
+      // create the restaurants feature layer using the service feature table
+      FeatureLayer restaurantLayer = new FeatureLayer(serviceFeatureTable);
+
+      // map the input fields in the feature layer to the dictionary symbol style's expected fields for symbols and text
+      HashMap<String, String> fieldMap = new HashMap<>();
+      fieldMap.put("healthgrade", "Inspection");
+
+      toggleGroup.selectedToggleProperty().addListener((observableValue, oldToggle, newToggle) -> {
+        if (newToggle == null)
+          toggleGroup.selectToggle(oldToggle);
+
+        // clear the existing dictionary symbol style and feature layer on the map
+        map.getOperationalLayers().clear();
+        dictionarySymbolStyle = null;
+        dictionaryRenderer = null;
+
+        // turn on the progress indicator when the symbols are loading
+        progressIndicator.setVisible(true);
+
+        if (toggleGroup.getSelectedToggle() == webStyleButton) {
+          dictionarySymbolStyle = new DictionarySymbolStyle(portalItem);
+          // load the symbol dictionary
+          dictionarySymbolStyle.loadAsync();
+          dictionarySymbolStyle.addDoneLoadingListener(() -> {
+            if (dictionarySymbolStyle.getLoadStatus() == LoadStatus.LOADED) {
+              // create a dictionary renderer with the dictionary symbol style
+              dictionaryRenderer = new DictionaryRenderer(dictionarySymbolStyle, fieldMap, fieldMap);
+              // renderer tells layer what symbols to apply to what features
```

Comment:
let's avoid the use of "tells", it's a bit too informal for doc language: how about the following

```suggestion
              // set the dictionary renderer as the feature layer's renderer to apply symbols to features
```